pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -150,6 +159,16 @@ func (rc *resourceMetricsCollector) collectNodeMemoryMetrics(ch chan<- metrics.M
 		metrics.NewLazyConstMetric(nodeMemoryUsageDesc, metrics.GaugeValue, float64(*s.Memory.WorkingSetBytes)))
 }
 
+func (rc *resourceMetricsCollector) collectContainerStartTime(ch chan<- metrics.Metric, pod summary.PodStats, s summary.ContainerStats) {
+	if s.StartTime.Unix() == 0 {
+		return
+	}
+
+	containerStartTimeSec := float64(s.StartTime.UnixNano()/int64(1e6)) / 1e3
```

Comment:
This construction seems a bit weird to me. 
We divide billionths by millionths, convert and then divide by thousandths.
Is there a reason for the multiple steps? We can see up a few lines that StartTime has a Unix() for the second, why are we calculating it ?

Reply:
I read, https://github.com/kubernetes/kubernetes/pull/102444#discussion_r654928012 but am still confused.

Reply:
Sorry, but metrics `container_start_time` is requested to second as unit but keep the millisecond resolution. and does this make sense?
```
containerStartTimeSec := float64(s.StartTime.UnixNano()/int64(time.Millisecond)) / 1e3
```

Reply:
Why not?
```
containerStartTimeSec := float64(s.StartTime.UnixNano()/int64(time.Second))
```

Reply:
This will lose the millisecond resolution result as `1.624396158e+09`, but required as format `1.624396158302e+09`.

Reply:
Right, sorry I forgot we want to have milisecond resolution

Have you tried converting to float first?
```
float64(s.StartTime.UnixNano()) / float64(time.Second)
```

Reply:
This works absolutely, and result as `1.6243962783020916e+09` which has precise to microsecond. If we do not mind it, this expression looks more simple and clear.

Reply:
LG